### PR TITLE
Install CUDA on Windows if cuda_compiler_version is present in %CONFIG%

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

## Questions

- [ ] `cuda_compiler_version` needs to be added to the Windows matrix in `conda-forge-pinning` -- see https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/914
- [ ] CUDA 9.2 requires `vs2015` compilers, but that will be addressed in `conda-forge-pinning` too? -- see https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/914#discussion_r515657282
- [ ] This kind of change deserves a new minor version or just patch? For now I just bumped the build number.
- [ ] There are some opportunities to speed up the CUDA installation, like installing only the needed components. I don't know which ones apply ([full list here](https://docs.nvidia.com/cuda/archive/11.0/cuda-installation-guide-microsoft-windows/index.html#install-cuda-software)), so for now all are being installed.
- [ ] Do we need to print a EULA statement somewhere in the CI logs?

### CODE REVIEW

**Comment by isuruf** [comment]
You'll have to apply the migration here manually (copy it to `.ci_support/migrations` and delete the timestamp) and rerender.

**Comment by jaimergp** [comment]
I am trying to rerender locally but the process deletes the manually added migration yaml right away and it's not applied. What's the procedure here?

**Comment by jaimergp** [comment]
@conda-forge-admin, please rerender

**Comment by isuruf** [comment]
https://github.com/conda-forge/conda-smithy/blob/master/conda_smithy/configure_feedstock.py#L1856 looks wrong. Maybe `if not isinstance(ts, (int, string, float)):`

**Comment by jaimergp** [comment]
I see. For now I am borrowing the  timestamp from the `cuda110` migration and that seems to make it survive the rerender, but the windows configs are not expanded with `cuda_compiler_version`, so I guess there's something wrong with the migrator:

```yaml
migrator_ts: 1601612527
__migrator:
  kind:
    version
  migration_number:
    1
  build_number:
    1

cuda_compiler:                 # [linux64 or win]
  - nvcc                       # [linux64 or win]

cuda_compiler_version:
  - None
  - 9.2                        # [linux64]
  - 10.0                       # [linux64 or win]
  - 10.1                       # [linux64 or win]
  - 10.2                       # [linux64 or win]
  - 11.0                       # [linux64 or win]
```

**Comment by isuruf** [comment]
> For now I am borrowing the timestamp from the cuda110 migration and that seems to make it survive the rerender,

The source of truth is taken to be the one in upstream conda-forge-pinning package, so, it'll not work.

**Comment by jaimergp** [comment]
Oh, yes, you are right, the logs confirm that. Then `conda-smithy` needs to be patched with that change in your comment?

**Comment by isuruf** [comment]
Yes

**Comment by jaimergp** [comment]
Ok, now it's working! Submitted a PR to `conda-smithy` as well.

This will need the CUDA 11 migration. Shall I add it here?

**Comment by isuruf** [comment]
> This will need the CUDA 11 migration. Shall I add it here?

Yes, please

**Comment by jaimergp** [comment]
Seems to be working now!

File: recipe/run_conda_forge_build_setup_win.bat [hunk 1]
```diff
@@ -51,6 +51,22 @@ if defined CI (
 :: Make paths like C:\hostedtoolcache\windows\Ruby\2.5.7\x64\bin garbage
 set "PATH=%PATH:ostedtoolcache=%"
 
+:: Install CUDA drivers if needed
+for %%i in ("%~dp0.") do set "SCRIPT_DIR=%%~fi"
+:: TODO: Remove before merge?
+conda.exe install -yq shyaml
```
Comment (isuruf) [inline recipe/run_conda_forge_build_setup_win.bat:57]: Move this to line 112 of  `.azure-pipelines/azure-pipelines-win.yml`.

**Comment by jaimergp** [comment]
Any other comments @conda-forge/conda-forge-ci-setup ?

**Review by isuruf — APPROVED** [review]
Looks good. Can you bump the version to 3.5.0?

File: .ci_support/migrations/cuda110.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+cuda_compiler_version:
+  - None
+  - 9.2                        # [linux64]
+  - 10.0                       # [linux64]
+  - 10.1                       # [linux64]
+  - 10.2                       # [linux64]
+  - 11.0                       # [linux64]
+
+# cdt_name:  # [os.environ.get("BUILD_PLATFORM", "").startswith("linux") or (os.environ.get("CONFIG_VERSION", "1") == "1" and linux)]
```
Comment (scopatz) [inline .ci_support/migrations/cuda110.yaml:20]: Should this block still be present?

**Review by scopatz — APPROVED** [review]

File: .ci_support/migrations/cuda110.yaml (hunk 2 above)
Comment (jaimergp) [reply .ci_support/migrations/cuda110.yaml:20]: This inherited from https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/884

**Comment by scopatz** [comment]
thanks @jaimergp!

**Pull request merged**